profiles/events.py

```diff
@@ -1,16 +1,59 @@
+import json
 import logging
-from typing import Callable
+from typing import Callable, Dict, List, Tuple
 
 from elife_bus_sdk.events import ProfileEvent
 from elife_bus_sdk.publishers import EventPublisher
-from flask import Flask
+from flask import Flask, url_for
+import requests
+from requests import RequestException
 
+from profiles.database import db
 from profiles.exceptions import UpdateEventFailure
 from profiles.models import Affiliation, EmailAddress, Profile
-
+from profiles.orcid import OrcidClient
 
 LOGGER = logging.getLogger(__name__)
 
+OPERATION_DELETE = 'delete'
+OPERATION_INSERT = 'insert'
+OPERATION_UPDATE = 'update'
+
+
+def maintain_orcid_webhook(orcid: Dict[str, str], orcid_client: OrcidClient) -> Callable[..., None]:
+    # pylint:disable=unused-argument
+    def webhook_maintainer(sender: Flask, changes: List[Tuple[db.Model, str]]) -> None:
+        profiles = list(filter(lambda x: isinstance(x[0], Profile) and x[0].orcid, changes))
+
+        if not profiles:
+            return
+
+        data = {
+            'client_id': orcid['client_id'],
+            'client_secret': orcid['client_secret'],
+            'scope': '/webhook',
+            'grant_type': 'client_credentials',
+        }
+
+        response = requests.post(url=orcid['token_uri'], data=data,
+                                 headers={'Accept': 'application/json'})
```

Comment:
Might be better to cache the results of this, we (should) only need to fetch one once.

Reply:
I agree. Probably best to have this as a stand alone task as it will effectively be setting up a caching solution that any part of the app could use. It may need more thought as to what other caching requirements we anticipate.